core/dbt/contracts/graph/manifest.py

```diff
@@ -946,11 +960,17 @@ def find_materialization_macro_by_name(
             and materialization_candidate.locality == Locality.Imported
             and core_candidates
         ):
-            deprecations.warn(
-                "package-materialization-override",
-                package_name=materialization_candidate.macro.package_name,
-                materialization_name=materialization_name,
-            )
+            # preserve legacy behaviour - allow materialization override
+            if get_flags().deprecate_package_materialization_builtin_override is False:
+                deprecations.warn(
+                    "package-materialization-override",
+                    package_name=materialization_candidate.macro.package_name,
+                    materialization_name=materialization_name,
+                )
+            else:
+                materialization_candidate = candidates.last_candidate(
+                    valid_localities=[Locality.Core, Locality.Root]
```

Comment:
was not able to do this lower-level using the `_find_macros_by_name` `filter` parameter because that filter only has a view of the macro candidate being iterated over. For this check, we need to exclude imported materialization macros only if the materialization is builtin in order to continue supporting custom materializations from packages (tested)